tests/address_bar_and_search/test_suggestion_engine_selection.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from selenium.webdriver import Firefox, Keys
+from selenium.webdriver.support import expected_conditions as EC
+
+from modules.browser_object import Navigation
+
+sites = ["Google", "Amazon.com", "Bing", "DuckDuckGo", "eBay"]
+
+
+@pytest.mark.parametrize("site", sites)
+def test_search_suggestion_for_engine_selection(driver: Firefox, site: str):
+    """
+    C1365228 - Test to verify that when entering "@", a list of search engines is suggested, and upon selecting an
+    engine, the search is performed using the selected engine.
+    """
+    nav = Navigation(driver).open()
+    nav.type_in_awesome_bar("@")
+
+    # Use base selector to find all search engine suggestions
```

Comment:
So, we're having flaky tests runs here, probably because the menu is taking a bit of time to populate. I recommend using some kind of iteration to `get_element` each of the target search engines individually before moving on. This will help because selenium `find_elements` doesn't wait for all possible matches to appear, it just returns when ANY elements match the selector. You could then _not_ use `get_elements` in the next line (if you collect the elements during iteration) OR you could keep that if that's your preferred way of collecting the elements.

Reply:
I've implemented a retry mechanism for locating search suggestion elements, not sure this is the best approach, I've run my tests several times without any failure though. Let me know if I should try a different approach on this one.

Reply:
This is still flaky for me (though quite a bit better than before).  You might try adding a new selector or changing the existing selector so that you can get the `data-l10n-args` attribute on the `.urlbarView-title` element. Presumably this command (don't use it exactly as written of course) would wait for your element to exist:
`nav.driver.find_element(By.CSS_SELECTOR, f'.urlbarView-title[data-l10n-args="{\"engine\":\"{site}\"}]')`

But that is pretty complex. You could change the check inside your iterator to
```python
nav.expect(
  EC.text_to_be_present_in_element(
    nav.get_element("search-suggestion-list"),
    site
  )
)
```